cpp/ql/src/semmle/code/cpp/ir/dataflow/internal/DataFlowPrivate.qll

```diff
@@ -393,11 +394,11 @@ private Instruction skipCopyValueInstructions(Instruction instr) {
 private predicate arrayReadStep(Node node1, ArrayContent a, Node node2) {
   a = TArrayContent() and
   // Explicit dereferences such as `*p` or `p[i]` where `p` is a pointer or array.
-  exists(LoadInstruction load, Instruction address |
-    load.getSourceValueOperand().isDefinitionInexact() and
-    node1.asInstruction() = load.getSourceValueOperand().getAnyDef() and
-    load = node2.asInstruction() and
-    address = skipCopyValueInstructions(load.getSourceAddress()) and
+  exists(LoadOperand operand, Instruction address |
+    operand.isDefinitionInexact() and
+    node1.asInstruction() = operand.getAnyDef() and
+    operand = node2.asOperand() and
+    address = skipCopyValueInstructions(operand.getUse().(LoadInstruction).getSourceAddress()) and
```

Comment:
```suggestion
    address = skipCopyValueInstructions(operand.getAddressOperand()) and
```
There are actually three different instructions that have `LoadOperand`s: `Load`, `ReturnValue`, and `ThrowValue`. It's probably good that this predicate now handles all three, but instead of relying on the use to be a `LoadInstruction`, you should use the simpler and more general `MemoryOperand::getAddressOperand()`.

Reply:
Cool. I wasn't aware of that predicate. That's really useful.

Reply:
Fixed in 3c778868f8a5142a6f1862167226b0609f7c1658. I had to modify the definition of `skipCopyValueInstructions` to take an `Operand` as an argument as there was no `getAddressOperandDef()`.